ciao-cli/main.go

```diff
@@ -175,6 +176,11 @@ func buildCiaoURL(format string, args ...interface{}) string {
 	return fmt.Sprintf(prefix+format, args...)
 }
 
+func buildBlockURL(format string, args ...interface{}) string {
+	prefix := fmt.Sprintf("https://%s:%d/v2/", *controllerURL, block.APIPort)
+	return fmt.Sprintf(prefix+format, args...)
+}
+
```

Comment:
I'm not really happy with this - we shouldn't be hardcoding our apis in our client in general, and it instead we should be discovering the URL. For an openstack api, you'd get the service endpoint from keystone, then build it from there. For a ciao api, we'd discover it from a "GET" on "/". However, these old openstack apis are really problematic from the get go, and we know they will eventually be phased out. I'd prefer an intermediate solution where we abstract the URL building from the rest of the client code, but knowing that @tmarcu is creating a new client, I think I'm ok with just doing the simplest fastest thing that gets us where we want to go in the short term. However I'm strongly against this approach as a long term thing.